Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -166,6 +176,23 @@ class LoginListViewController: UIViewController {
     }
 }
 
+// MARK: - Favicon Loading
+extension LoginListViewController {
+
+    private func loadFaviconsForLogins(logins: [Login]) -> Deferred<Maybe<[GUID: [Favicon]]>> {
+        let urls = logins.map { $0.hostname }
+        return self.profile.favicons.getFaviconsForHistoryURLs(urls) >>== { favicons in
+            let filteredFavicons = favicons.asArray().flatMap { $0 }
+            var loginFaviconMap = [GUID: [Favicon]]()
+            logins.forEach { login in
+                // Filter by login hostname -> history URL and extract out the favicon
+                loginFaviconMap[login.guid] = filteredFavicons.filter { $0.0.hasPrefix(login.hostname) } .map { $0.1 } ?? []
```

Comment:
This query made me feel gross writing it. On top of this awful searching the `getFaviconsForHistoryURLs` query attaches a ton of LIKE statements to bulk find all favicons for the given history URLs. The biggest problem I ran into was being able to link a login entry with a favicon since a login has no identifying property to connect the two. I was trying to avoid having to attach additional information to the login table just to find a favicon considering a login shouldn't have to know about favicons. You also mentioned earlier that they don't even live in the same database so that adds another layer of complexity.

Unless you have any ideas about another way to simplify this without schema changes I don't know if I feel comfortable shipping this favicon branch for the 2.0 release. I'd prefer to take this back to the drawing board for a better solution for 3.0 and go with favicon-less logins for now.